src/lib/Web3Function.ts

```diff
@@ -38,58 +38,54 @@ export class Web3Function {
   private async _onFunctionEvent(
     event: Web3FunctionEvent
   ): Promise<Web3FunctionEvent> {
-    switch (event?.action) {
-      case "start": {
-        const prevStorage = { ...event.data.context.storage };
-
-        try {
-          const { result, ctxData } = await this._run(event.data.context);
-
-          const difference = diff(prevStorage, ctxData.storage);
-          for (const key in difference) {
-            if (difference[key] === undefined) {
-              difference[key] = null;
-            }
+    if (event?.action === "start") {
+      const prevStorage = { ...event.data.context.storage };
+
+      try {
+        const { result, ctxData } = await this._run(event.data.context);
+
+        const difference = diff(prevStorage, ctxData.storage);
+        for (const key in difference) {
+          if (difference[key] === undefined) {
+            difference[key] = null;
           }
+        }
+
+        const state = Object.keys(difference).length === 0 ? "last" : "updated";
 
-          const state =
-            Object.keys(difference).length === 0 ? "last" : "updated";
-
-          return {
-            action: "result",
-            data: {
-              result,
-              storage: {
-                state,
-                storage: ctxData.storage,
-                diff: difference,
-              },
+        return {
+          action: "result",
+          data: {
+            result,
+            storage: {
+              state,
+              storage: ctxData.storage,
+              diff: difference,
             },
-          };
-        } catch (error) {
-          return {
-            action: "error",
-            data: {
-              error: {
-                name: error.name,
-                message: `${error.name}: ${error.message}`,
-              },
-              storage: {
-                state: "last",
-                storage: prevStorage,
-                diff: {},
-              },
+          },
+        };
+      } catch (error) {
+        return {
+          action: "error",
+          data: {
+            error: {
+              name: error.name,
+              message: `${error.name}: ${error.message}`,
             },
-          };
-        } finally {
-          this._exit();
-        }
-        break;
+            storage: {
+              state: "last",
+              storage: prevStorage,
+              diff: {},
+            },
+          },
+        };
+      } finally {
+        this._exit();
       }
-      default:
-        Web3Function._log(`Unrecognized parent process event: ${event.action}`);
-        throw new Error(`Unrecognized parent process event: ${event.action}`);
     }
+
```

Comment:
should this be in an else block? 


```ts
      finally {
        this._exit();
      }
```
This exits with a setTimeout which might cause the error throwing to trigger even when `event?.action === "start"`

Reply:
thanks @brandonchuah, fixed in f4b838329e316bd42169c469a86abd891051b7ec